enumerable_methods.rb

```diff
@@ -0,0 +1,156 @@
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    for i in self
+      yield i
+    end
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    ind = 0
+    for i in self
+      yield(i, ind)
+      ind += 1
+    end
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    new_arr = []
+    my_each do |n|
+      new_arr << n if yield(n)
+    end
+    new_arr
+  end
+
+  def my_all?
+    # return true if empty array is passed, regardless of block_given
+    return true if (self.class == Array && count.zero?) || (!count.zero? && !block_given?)
+
+    my_each do |n|
+      return false unless yield(n)
+    end
+    true
+  end
+
+  def my_any?
+    return false if self.class == Array && count.zero?
+    return true if !count.zero? && !block_given?
+
+    my_each do |n|
+      return true if yield(n)
+    end
+    false
+  end
```

Comment:
- [ ] should have working `my_any?` identical and returns the same thing as ruby's [`any?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-any-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_any.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

 #### my_any? 

- [ ] when no block or argument is given returns true if at least one of the collection is not false or nil

```ruby     
[false, nil].my_any? == [false, nil].any? => false
```
- Yours returns the wrong answer
![image](https://user-images.githubusercontent.com/52209067/80322442-75fb8d00-87e2-11ea-898d-1da44ba3b327.png)

- Real any?
![image](https://user-images.githubusercontent.com/52209067/80322448-83187c00-87e2-11ea-8346-83100e5898b6.png)

- [ ] when a class is passed as an argument returns true if at least one of the collection is a member of such class

```ruby
[1, 2.5, 'a', 9].my_any?(Integer) == [1, 2.5, 'a', 9].any?(Integer) => true
```
- Yours is retrieving an ArgumentError
![image](https://user-images.githubusercontent.com/52209067/80322459-a2170e00-87e2-11ea-92c1-337d77a761bd.png)

- Real any?
![image](https://user-images.githubusercontent.com/52209067/77689595-8f04ea00-6f67-11ea-9f51-4536c3f8049b.png)

- [ ] when a Regex is passed as an argument returns false if none of the collection matches the Regex

```ruby
%w[dog door rod blade].my_any?(/z/) == %w[dog door rod blade].any?(/z/) => false
```
- Yours retrieves an ArgumentError
![image](https://user-images.githubusercontent.com/52209067/80322468-b824ce80-87e2-11ea-9306-23edd8bf0e10.png)

- Real any?
![image](https://user-images.githubusercontent.com/52209067/77689742-cf646800-6f67-11ea-9e3e-eec7a960fe38.png)

- [ ] when a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern

```ruby
[3,4,7,11].my_any?(3) should be equal to this [3,4,7,11].any?(3) => true
```
- Yours retrieves an ArgumentError
![image](https://user-images.githubusercontent.com/52209067/80322486-d4287000-87e2-11ea-89ca-2c2624ebc1ea.png)

- Real any?
![image](https://user-images.githubusercontent.com/52209067/77689883-10f51300-6f68-11ea-8a4c-319977bb3842.png)